frontend/components/Modal/Modal.js

```diff
@@ -0,0 +1,81 @@
+import React, { useLayoutEffect, useState } from "react";
+import { createPortal } from "react-dom";
+import styled from "@emotion/styled";
+import Button from "../Button/Button";
+import Flex from "../Flex/Flex";
+import Text from "../Text/Text";
+import { color, spacing } from "../../theme";
+
+const StyledModalBackground = styled.div`
+  position: fixed;
+  inset: 0;
+  background-color: ${color.black60};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledModal = styled.div`
+  padding: ${spacing["2"]} ${spacing["2.25"]};
+  background-color: ${color.white};
+  min-width: 400px;
+  border-radius: ${spacing["0.75"]};
+`;
+
+const Modal = ({
+  title,
+  children,
+  onClose,
+  onSubmit,
+  cancelText = "Cancel",
+  submitText = "Submit",
+}) => {
+  const [appRoot, setAppRoot] = useState(null);
+  useLayoutEffect(() => {
+    if (appRoot !== document?.getElementById("app-root")) {
+      setAppRoot(document.getElementById("app-root"));
+    }
+  }, [appRoot]);
```

Comment:
Please elaborate on the approach used here

Reply:
I originally had a useLayoutEffect here to prevent the modal from rendering if the div with id, "app-root" was not rendered in the unit tests. However, I was able to find a way to render the div before rendering the component in the unit tests.